tests/framework/helpers/JsonTest.php

```diff
@@ -156,10 +156,11 @@ public function testHtmlEncode()
 
     public function testDecode()
     {
-        // empty value
+        // empty values
         $json = '';
-        $actual = Json::decode($json);
-        $this->assertNull($actual);
+        $this->assertSame('', Json::decode($json));
```

Comment:
While this is more logical, this is breaking existing behavior that may be relied on.

Reply:
Breaks BC? | ✔️

Reply:
Why do you think it is logical? Empty string is invalid JSON, the only logical solution is throwing exception in that case. Replacing one invalid result with another invalid result is definitely not worth the BC break (and `null` is at least consistent with default PHP behavior).

Reply:
@rob006 because `@return string`

Reply:
> because `@return string`

Then fix the phpdoc.

> `json_decode(json_encode('')) === ''`

This example is completely unrelated. `json_encode('')` returns `""` (valid JSON representing empty string), while your PR changes behavior for empty string (invalid JSON - `json_decode('')` returns `null` and throws error when you use `JSON_THROW_ON_ERROR` as flag).

Reply:
@rob006 
> (invalid JSON - json_decode('') returns null and throws error when you use JSON_THROW_ON_ERROR as flag

I can't repeat this - https://3v4l.org/IRHM4

Reply:
> I can't repeat this - https://3v4l.org/IRHM4

Then you should probably "review standard before writing something like this": https://www.php.net/manual/en/function.json-decode.php - flags are passed as 4th argument: https://3v4l.org/92N19

Reply:
@rob006 is right. Empty string is not a valid JSON.

Reply:
@samdark

> review standard before writing something like this

https://www.json.org/json-en.html, [more info](https://www.crockford.com/mckeeman.html)

Reply:
There's a huge difference between empty PHP string and empty JSON string (`'""'`). Try https://jsonlint.com/. Enter nothing, press "Validate JSON".

Reply:
@rob006 yes, my fail - I'm copied arguments from `json_encode()`, but existed code don't throw exception in this case too

Reply:
@samdark in any case, you forget that this behavior already existed in current version

https://github.com/yiisoft/yii2/blob/c8e4e0727a122fddcd021f4419b939dc841d769c/framework/helpers/BaseJson.php#L117

Reply:
That's a bit different behavior. It is returning `null` for non-valid JSON, it's not returning empty string in any case.

Reply:
@samdark
> It is returning null for non-valid JSON

Current behavior too wrong. Method must return null or throw exception for **any** invalid JSON, absolutely.